src/main/scala/tech/cryptonomic/conseil/routes/Tezos.scala

```diff
@@ -35,79 +33,86 @@ object Tezos {
-    provide(keyStore)
+      val keyStore = KeyStore(publicKey = publicKey, privateKey = privateKey, publicKeyHash = publicKeyHash)
+      provide(keyStore)
   }
 
+  def apply(implicit apiExecutionContext: ExecutionContext) = new Tezos(ApiNetworkOperations())
+
 }
 
 /**
   * Tezos-specific routes.
   * The mixed-in [[DatabaseApiFiltering]] trait provides the
   * instances of filtering execution implicitly needed by
   * several Api Operations, based on database querying
+  *
   * @param apiExecutionContext is used to call the async operations exposed by the api service
   */
-class Tezos(implicit apiExecutionContext: ExecutionContext) extends LazyLogging with DatabaseApiFiltering with RouteHandling {
+class Tezos(apiNetworkOperations: ApiNetworkOperations)
+  (implicit apiExecutionContext: ExecutionContext) extends LazyLogging with RouteHandling {
 
   import Tezos._
 
   /*
    * reuse the same context as the one for ApiOperations calls
    * as long as it doesn't create issues or performance degradation
    */
-  override val asyncApiFiltersExecutionContext = apiExecutionContext
+  val asyncApiFiltersExecutionContext = apiExecutionContext
 
   /** expose filtered results through rest endpoints */
   val route: Route = pathPrefix(Segment) { network =>
-    get {
-      gatherConseilFilter{ filter =>
-        validate(filter.limit.forall(_ <= 10000), "Cannot ask for more than 10000 entries") {
-          pathPrefix("blocks") {
-            pathEnd {
-              completeWithJson(ApiOperations.fetchBlocks(filter))
-            } ~ path("head") {
-                completeWithJson(ApiOperations.fetchLatestBlock())
-            } ~ path(Segment).as(BlockHash) { blockId =>
-                complete(
-                  handleNoneAsNotFound(ApiOperations.fetchBlock(blockId))
-                )
-            }
-          } ~ pathPrefix("accounts") {
-            pathEnd {
-              completeWithJson(ApiOperations.fetchAccounts(filter))
-            } ~ path(Segment).as(AccountId) { accountId =>
-              complete(
-                handleNoneAsNotFound(ApiOperations.fetchAccount(accountId))
-                )
-            }
-          } ~ pathPrefix("operation_groups") {
-            pathEnd {
-              completeWithJson(ApiOperations.fetchOperationGroups(filter))
-            } ~ path(Segment) { operationGroupId =>
-              complete(
-                handleNoneAsNotFound(ApiOperations.fetchOperationGroup(operationGroupId))
-              )
-            }
-          } ~ pathPrefix("operations") {
-            path("avgFees") {
-                complete(
-                  handleNoneAsNotFound(ApiOperations.fetchAverageFees(filter))
-                )
-            } ~ pathEnd {
-                completeWithJson(ApiOperations.fetchOperations(filter))
+    apiNetworkOperations.getApiOperations(network) { apiOperations =>
+      apiNetworkOperations.getApiFiltering(network, asyncApiFiltersExecutionContext) { apiFiltering =>
```

Comment:
Maybe this might change later, but just as a side-note here... if you make
`...getApiFiltering(...) { implicit apiFiltering => ... }`

will the implicit instances in the filtering be available already in scope for the `apiOperations` methods?

So you won't need to later `import apiFiltering._`...

Reply:
it won't work, because implicit instances are inside the object